### PR TITLE
Replace native textarea with contentEditable for inline autocomplete

### PR DESCRIPTION
**Pull request by vimzh**
## Summary

Replaces the native `<textarea>` in `AutoCompleteTextarea` with a `contentEditable` div (`ContentEditableTextArea`) to enable inline ghost text rendering for autocomplete suggestions.

### Review feedback addressed
- Removed unused `_cursorToLinePos` function and all dead autocomplete state (`ghostPos`, `cursorAtFetch`, `isTyped`, `debouncedText`)
- Replaced `.at(-1)` with bracket access for ES2020 compatibility
- Fixed broken input logic — complete rewrite of contentEditable input handling

### What changed

**ContentEditableTextArea (new component):**
- All input routed through React state — no browser DOM mutations
- Cursor position preserved across re-renders via `pendingCaretRef`
- Full selection support (select-all + Backspace/Delete/type)
- Modifier key support: Option+Backspace/Delete (word), Cmd+Backspace/Delete (line)
- Cmd+Z/Redo blocked (would desync React state)
- IME composition guards for CJK input
- Grapheme-safe deletion for emoji via `Intl.Segmenter`
- Cut handler (clipboard write + state delete)
- Drag-and-drop blocked
- Spell-check replacements handled (`insertReplacementText`)
- `{...rest}` spread before explicit handlers to prevent override
- Placeholder alignment matched to `inputVariants` padding
- Zero-width space stripped from text extraction

**AutocompleteTextArea (simplified):**
- Removed dead imports (`useAutocomplete`, `useConfiguration`)
- `isActive` prop properly gates ghost text display
- `aria-invalid` and `aria-describedby` now forwarded
- Cached `text.split('\n')` (was called 3x per render)

**Tests:**
- 24 unit tests covering hook behavior, rendering, accessibility, controlled values, and edge cases

## Test plan

- [x] TypeScript: zero errors (`npx tsc --noEmit`)
- [x] Unit tests: 24/24 passing
- [x] Full design-system suite: 33/33 passing
- [x] Prettier: all files formatted
- [x] No unused imports, no dead code, no `any` types
- [x] Consumer compatibility verified (ContentEditorTextArea, AutocompletionSection)
- [x] Barrel exports intact (all 4 components)
- [x] Package build passes
- [x] Security review: no XSS vectors, paste uses text/plain only
- [x] Manual test: type, Enter, Backspace, select-all+delete, Option+Backspace, Cmd+X, paste

### CODE REVIEW

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 1]
```diff
@@ -185,183 +39,132 @@ export const AutoCompleteTextarea: FC<AutocompleteTextAreaProps> = ({
...
-          {
-            onSuccess: (data: AutocompleteResponse) => {
-              setSuggestion(data.data?.autocompletion ?? '');
-              setCursorAtFetch(cursor);
-            },
-          }
-        );
-      } catch (err) {
-        console.error('Autocomplete error:', err);
+  const _cursorToLinePos = useCallback((cursor: number, src: string) => {
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx]: dont seems to be used

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 2]
```diff
@@ -185,183 +39,132 @@ export const AutoCompleteTextarea: FC<AutocompleteTextAreaProps> = ({
...
     }, 0);
-  };
+  }, [text, suggestion, suggestionProp, cursorAtFetch]);
+
+  const activeGhost = suggestionProp ?? (suggestion || undefined);
+  const activeLine = suggestionProp
+    ? text.split('\n').length - 1
+    : ghostPos?.line;
+  const activeOffset = suggestionProp
+    ? (text.split('\n').at(-1)?.length ?? 0)
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx]: Property 'at' does not exist on type 'string[]'

**Review by aymericzip — COMMENTED** [review]
Hey @vimzh 

Thanks for your contribution, 

Here a quick review

**Comment by aymericzip** [comment]
the logic is broken 


https://github.com/user-attachments/assets/8c0ce25c-dd5d-4776-b47e-736ae04f7cca

**Comment by vimzh** [comment]
Hey @aymericzip, thanks for the review and for recording the video, I've pushed a fix addressing all the feedback
Let me know if there's anything else you'd like me to adjust!

**Comment by vimzh** [comment]
Hey @aymericzip, thanks for the review! I've pushed the changes:

- Removed all useCallback/memo wrappers (React compiler handles memoization)
- Removed Line.displayName (React 18 pattern)
- Imports were already using named imports (import { FC } from 'react')

Let me know if there's anything else!

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 3]
```diff
@@ -39,329 +15,97 @@ export const useDebounce = <T,>(value: T, delay: number): T => {
...
- * />
- * ```
- *
- * ## Accessibility
- * - Ghost layer is properly hidden from screen readers
- * - Maintains focus management during suggestion acceptance
- * - Preserves keyboard navigation patterns
- * - Respects reduced motion preferences
- */
 export const AutoCompleteTextarea: FC<AutocompleteTextAreaProps> = ({
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx:62]: restore jsdoc

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 4]
```diff
@@ -39,329 +15,97 @@ export const useDebounce = <T,>(value: T, delay: number): T => {
...
-  const placeholderRef = useRef<HTMLSpanElement>(null);
-  const ghostLayerRef = useRef<HTMLDivElement>(null);
-  const [suggestionPosition, setSuggestionPosition] = useState<{
-    left: number;
-    top: number;
-  } | null>(null);
-  const [cursorAtFetch, setCursorAtFetch] = useState(-1);
-
-  // Only update this “debouncedText” after the user stops typing for 200ms
-  const debouncedText = useDebounce(text, 200);
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx:198]: useDebouce removed, 

Dont seems to be used anymore

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 5]
```diff
@@ -39,329 +15,97 @@ export const useDebounce = <T,>(value: T, delay: number): T => {
...
-        const contextBeforeLines = beforeLines.slice(
-          Math.max(0, beforeLines.length - numLines - 1),
-          -1
-        );
-        const contextBefore = contextBeforeLines.join('\n');
-        const currentLine = beforeLines[beforeLines.length - 1] ?? '';
-        const afterLines = after.split('\n');
-        const contextAfter = afterLines.slice(1, numLines + 1).join('\n');
-
-        autocomplete(
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx:226]: autocomplete removed. can you explain how work the new one ?

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 6]
```diff
@@ -39,329 +15,97 @@ export const useDebounce = <T,>(value: T, delay: number): T => {
...
+      value={text}
+      onChange={(val) => {
+        setText(val);
+        setSuggestion('');
+
+        if (props.onChange) {
+          const evt = {
+            target: { value: val },
+            currentTarget: { value: val },
+          } as React.ChangeEvent<HTMLTextAreaElement>;
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx]: import { ChangeEvent } from 'react'

(codebase convention)

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx [hunk 7]
```diff
@@ -39,329 +15,97 @@ export const useDebounce = <T,>(value: T, delay: number): T => {
...
+          props.onChange(evt);
+        }
+      }}
+      onKeyDown={(e) => {
+        if (e.key === 'Tab' && (suggestionProp ?? suggestion)) {
+          e.preventDefault();
+          acceptSuggestion();
+        }
+        props.onKeyDown?.(
+          e as unknown as React.KeyboardEvent<HTMLTextAreaElement>
```
Comment (aymericzip) [inline packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx]: import { KeyboardEvent } from 'react'

File: packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx (hunk 5 above)
Comment (vimzh) [reply packages/@intlayer/design-system/src/components/TextArea/AutocompleteTextArea.tsx:226]: The old API-based autocomplete was already disabled with a TODO comment so we removed that dead code. The main change is the rendering layer - instead of a native textarea with an absolute-positioned ghost overlay, we now use a contentEditable div that renders ghost text inline as a span right after the cursor. The suggestion prop on AutoCompleteTextarea is how you feed in suggestions from any source, and the user accepts them with Tab. Whenever you’re ready to bring back the AI autocomplete, you just need to wire the fetcher to that prop.

**Comment by aymericzip** [comment]
good job! thanks @vimzh
